src/user_input/mod.rs

```diff
@@ -111,48 +111,7 @@ mod trait_serde;
 pub mod updating;
 
 /// A trait for defining the behavior expected from different user input sources.
-///
-/// Implementers of this trait should provide methods for accessing and processing user input data.
-///
-/// # Examples
-///
-/// ```rust
-/// use std::hash::{Hash, Hasher};
-/// use bevy::prelude::*;
-/// use bevy::math::{Vec2, FloatOrd};
-/// use serde::{Deserialize, Serialize};
-/// use leafwing_input_manager::prelude::*;
-/// use leafwing_input_manager::axislike::{DualAxisType};
-/// use leafwing_input_manager::clashing_inputs::BasicInputs;
-///
-/// #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Reflect, Serialize, Deserialize)]
-/// pub struct MouseScrollAlwaysFiveOnYAxis;
-///
-/// // Add this attribute for ensuring proper serialization and deserialization.
-/// #[serde_typetag]
-/// impl UserInput for MouseScrollAlwaysFiveOnYAxis {
-///     fn kind(&self) -> InputControlKind {
-///         // Returns the kind of input this represents.
-///         //
-///         // In this case, it represents an axial input.
-///         InputControlKind::Axis
-///     }
-///
-///     fn decompose(&self) -> BasicInputs {
-///         // Gets the most basic form of this input for clashing input detection.
-///         //
-///         // This input is not buttonlike, so it uses `None`.
-///         BasicInputs::None
-///     }
-/// }
-///
-/// // Remember to register your input - it will ensure everything works smoothly!
-/// let mut app = App::new();
-/// app.register_user_input::<MouseScrollAlwaysFiveOnYAxis>();
-/// ```
-pub trait UserInput:
```

Comment:
I think we can add a detailed example.rs for this

Reply:
Agreed!